servers/dsub/jobs/controllers/aggregation_controller.py

```diff
@@ -24,47 +19,110 @@ def get_job_aggregations(timeFrame, projectId=None):
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=7),
-            StatusCount(status='Failed', count=4)
-        ]))
-
-    ownerAggregation = Aggregation(
-        name='AnotherLabel', key='job-id', entries=[entry1, entry2])
-
-    projectAggregation = Aggregation(
-        name='User', key='userId', entries=[userEntry1, userEntry2])
+    window_min = time_frame.time_frame_to_start_time(timeFrame)
+    provider = providers.get_provider(_provider_type(), projectId,
+                                      _auth_token())
+
+    jobs = jobs_generator.generate_jobs_by_window(provider, projectId,
+                                                  window_min)
+
+    total_summary = {}
+    user_summary = {}
+    job_name_summary = {}
+    label_summaries = {}
+    label_freq = {}
+
+    for job in jobs:
+        _count_total_summary(job, total_summary)
+        _count_for_key(job, user_summary, lambda j: j.extensions.user_id)
+        _count_for_key(job, job_name_summary, lambda j: j.name)
+        _count_top_labels(label_summaries, job, label_freq)
+
+    aggregations = [
+        _to_aggregation('User Id', 'userId', user_summary),
+        _to_aggregation('Job Name', 'name', job_name_summary)
+    ]
+
+    _to_top_labels_aggregations(label_freq, label_summaries, aggregations)
 
     return AggregationResponse(
-        summary=summaryCounts,
-        aggregations=[ownerAggregation, projectAggregation])
+        summary=_to_summary_counts(total_summary), aggregations=aggregations)
+
+
+def _count_total_summary(job, total_summary):
+    if job.status not in total_summary:
+        total_summary[job.status] = 0
+    total_summary[job.status] += 1
+
+
+def _count_for_key(job, summary, key_lambda):
+    key = key_lambda(job)
+
+    count = summary.get(key, {})
+
+    if job.status not in count:
+        count[job.status] = 0
+    count[job.status] += 1
+
+    summary[key] = count
+
+
+def _count_top_labels(label_summaries, job, label_freq):
+    for label in job.labels:
+        label_freq[label] = label_freq.get(label, 0) + 1
+
+        label_summary = label_summaries.get(label, {})
+        _count_for_key(job, label_summary, lambda j: j.labels[label])
+        label_summaries[label] = label_summary
+
+
+def _to_summary_counts(summary_counts):
+    return StatusCounts([
+        StatusCount(status, count) for status, count in summary_counts.items()
+    ])
+
+
+def _to_aggregation(name, key, summary):
+    entries = []
+
+    for item, counts_dict in summary.items():
+        counts_list = []
+
+        for status, count in counts_dict.items():
+            counts_list.append(StatusCount(status=status, count=count))
+
+        entries.append(
+            AggregationEntry(
+                label=item, status_counts=StatusCounts(counts_list)))
+
+    return Aggregation(name=name, key=key, entries=entries)
+
+
+def _to_top_labels_aggregations(label_freq, label_summaries, aggregations):
```

Comment:
There should be a comment somewhere in this file that describes the approach we're using here and why we're doing it.

@bfcrampton FYI Zach and I came up with this as an initial solution: basically pick out the top 3 highest frequency labels and aggregate by them. Later we might make this configurable, or just defer to the user to pick better labels.

Reply:
Seems like a good first approach to me.